multicluster/service-mirror/cluster_watcher.go

```diff
@@ -860,6 +886,11 @@ func (rcsw *RemoteClusterServiceWatcher) Start(ctx context.Context) error {
 			case <-ticker.C:
 				ev := RepairEndpoints{}
 				rcsw.eventsQueue.Add(&ev)
+			case alive := <-rcsw.liveness:
+				rcsw.log.Debugf("gateway liveness change from %t to %t", rcsw.alive, alive)
+				rcsw.alive = alive
+				ev := RepairEndpoints{}
+				rcsw.eventsQueue.Add(&ev)
```

Comment:
This is mutating shared state so some locking is required?

Reply:
I thought about that but decided that it's not an issue if cluster watcher methods see an old `alive` value.

My reasoning is that `alive` is only mutated here, but is read in other methods. Those methods are only called when processing an event. So, if the watcher is processing an event and sees an old `alive` value, we still will queue a `RepairEndpoints` event here; the watcher will have to process this new event where it sees the updated `alive` value.

Reply:
Agreed on the scenario, I'm just thinking on following [go docs](https://go.dev/ref/mem) advise to avoid having weird scenarios come up when things get more complex down the road: "Programs that modify data being simultaneously accessed by multiple goroutines must serialize such access". This is just a TIOLI though.

Reply:
> This is just a TIOLI though.

It's definitely a good conversation to have no worries.

So I see the point about the value of `alive` being observed simultaneously—first from this goroutine where we are selecting over the ticker/liveness, and the second from the other goroutine responsible for calling `processEvents`.

Looking through the docs you linked, it's not an issue if we observe old `alive` values because a `RepairEndpoints` event still gets queued which will observe the new value. I can't really think of a scenario where'd we end up with an error after reading an old value.

I likely just stated what you're already noting on but just wanted to make sure were on the same page. I'd be open to adding synchronization, but it felt safe enough to avoid it here—maybe those are famous last words? If you can go either on way on this I'll probably keep it out for now, but curious to see what other reviewers think.

Reply:
I definitely can go either way on this. The changes tests pretty good also and I don't wanna block on this, so giving it a :heavy_check_mark: